llmproxy/models/vertexai.py

```diff
@@ -0,0 +1,52 @@
+from google.cloud import aiplatform
+from google.auth import exceptions
+from llmproxy.models.base import BaseChatbot, CompletionResponse
+from llmproxy.utils.enums import BaseEnum
+from llmproxy.utils.log import logger
+from vertexai.language_models import TextGenerationModel
+
+class VertexAIModel(str, BaseEnum):
+    PALM_TEXT = "text-bison@001"
+    PALM_CHAT = "chat-bison"
+
+class VertexAI(BaseChatbot):
+    def __init__(
+        self,
+        prompt: str = "",
+        temperature: float = 0,
+        model: VertexAIModel = VertexAIModel.PALM_TEXT,
+        project_id: str | None = "",
+        location: str = "",
+    ) -> None:
+        self.prompt = prompt
+        self.temperature = temperature
+        self.model = model
+        aiplatform.init(project=project_id, location=location)
```

Comment:
If passing in a location that DNE, then you should have a check/default value for that

Reply:
default value will be an empty string